packages/stockflux-launcher/src/App.js

```diff
@@ -1,66 +1,144 @@
 } from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
 import ToolBar from './toolbar/ToolBar';
 import CloseButton from './toolbar/CloseButton';
 import './App.css';
 
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
-
 export default () => {
+  const [alignment, setAlignment] = useState('horizontal');
+  const [isCloudMode, setCloudMode] = useState(false);
+  const [isDocked, setDocked] = useState(true);
+  const [options] = useOptions();
+  const [undockWidth, setUndockWidth] = useState(1000);
+  const [undockHeight, setUndockHeight] = useState(50);
+  const [undockTop] = useState(50);
+  const [undockLeft, setUndockLeft] = useState(0);
+
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
-    OpenfinApiHelpers.getCurrentWindowSync(),
+    window.fin.Window.getCurrentSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    { dockedWidth: 50, dockedHeight: 50 },
+    {
+      undockPosition: { top: undockTop, left: undockLeft },
+      undockSize: { width: undockWidth, height: undockHeight }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  let currentDirection = edge !== ScreenEdge.NONE ? edge : ScreenEdge.TOP;
+  let left =
+    window.screenLeft >= window.screen.availWidth ? window.screenLeft : 0;
+  let right =
+    window.screenLeft >= window.screen.availWidth
+      ? window.screenLeft
+      : window.screen.availWidth - 50;
+  let midpoint =
+    left === 0
+      ? (left + right) / 4
+      : window.screen.availWidth + (left + right) / 8;
+
+  // Handle initialDocked false resize to mini window
+  useEffect(() => {
+    if (
+      options &&
+      options.customData &&
+      options.customData.initialDocked === false
+    ) {
+      setUndockLeft(midpoint);
+      windowActions.dockNone();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
+
+  useEffect(() => {
+    if (edge !== ScreenEdge.NONE) {
+      setAlignment(edge !== ScreenEdge.TOP ? 'vertical' : 'horizontal');
+    }
+  }, [edge]);
+
+  useEffect(() => {
+    setCloudMode(
+      options && options.customData && options.customData.cloudMode === true
+    );
+  }, [options, setCloudMode]);
+
   useEffect(() => {
-    prevEdgeRef.current = edge;
-  });
-  const prevEdge = prevEdgeRef.current;
+    setDocked(!isCloudMode && edge !== ScreenEdge.NONE);
+  }, [edge, isCloudMode]);
 
-  const edgeToBeChecked = edge === ScreenEdge.NONE ? prevEdge : edge;
+  const undock = async () => {
+    switch (currentDirection) {
+      case ScreenEdge.LEFT:
+        setUndockHeight(600);
+        setUndockWidth(50);
+        setUndockLeft(left);
+        break;
+      case ScreenEdge.RIGHT:
+        setUndockHeight(600);
+        setUndockWidth(50);
+        setUndockLeft(right);
+        break;
+      case ScreenEdge.TOP:
+        setUndockHeight(50);
+        setUndockWidth(1000);
+        setUndockLeft(midpoint);
```

Comment:
When undocking the screen from a secondary monitor the position of the hover launcher resets to the main display. 
This only occurs on secondary monitors which are to the negative direction of the primary display.

Reply:
The undock should now work for all monitor setups, including 1 | 2, 2 | 1 and vertical alignments 1 / 2 and 2 / 1. The undock will move to the top left of that monitor and allow dragging